### PR TITLE
EMA & gradient accumulation & ConvNeXt

### PR DESCRIPTION
**Pull request by flytocc**

### CODE REVIEW

**Comment by flytocc** [comment]
| 模型      | top1 acc (参考精度) | top1 acc (复现精度) | 权重 \| 训练日志 |
|:---------:|:------:|:----------:|:----------:|
| convnext_tiny | 0.82060   | 0.82056   | checkpoint-best.pd \| log.txt |

权重及训练日志下载地址：[百度网盘](https://pan.baidu.com/s/1zJENrWFRXSFwIPdxt-QMIA?pwd=jsqp)

ps: 1. 精度 82.056 是在这个[repo](https://github.com/flytocc/ConvNeXt-paddle)训出来的;
2. PaddleClas可加载 checkpoint_for_paddleclas_82.030.params 验证（精度较低是tansforms不同造成的）。

File: ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# global configs
```
Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: yaml名字和模型名字保持一致

Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: yaml名字和模型名字保持一致

File: ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  eval_interval: 1
+  epochs: 300
+  print_batch_step: 10
+  use_visualdl: False
+  # used for static mode and model export
+  image_shape: [3, 224, 224]
+  save_inference_dir: ./inference
+  # training model under @to_static
+  to_static: False
+  update_freq: 8  # for 8 cards
```
Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 梯度累加的功能是否有验证呢？

Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 梯度累加的功能是否有验证呢？

File: ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  name: AdamW
+  beta1: 0.9
+  beta2: 0.999
+  epsilon: 1e-8
+  weight_decay: 0.05
+  one_dim_param_no_weight_decay: True
+  lr:
+    # for 8 cards
+    name: Cosine
+    learning_rate: 4e-3  # lr 4e-3 for total_batch_size 4096
```
Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 现在的bs和lr应该没有对应上，下边的bs是128，卡数是8的话总bs应该是1024，lr应该是1e-3？

Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 现在的bs和lr应该没有对应上，下边的bs是128，卡数是8的话总bs应该是1024，lr应该是1e-3？

File: ppcls/data/preprocess/ops/operators.py [hunk 4]
```diff
@@ -102,28 +103,59 @@ class OperatorParamError(ValueError):
 class DecodeImage(object):
     """ decode image """
 
-    def __init__(self, to_rgb=True, to_np=False, channel_first=False):
+    def __init__(self,
+                 to_rgb=True,
+                 to_np=False,
+                 channel_first=False,
+                 backend="cv2",
+                 return_numpy=True):
```
Comment (cuicheng01) [inline ppcls/data/preprocess/ops/operators.py]: 改这个文件的目的是？

Comment (cuicheng01) [inline ppcls/data/preprocess/ops/operators.py]: 改这个文件的目的是？

Comment (flytocc) [reply ppcls/data/preprocess/ops/operators.py]: 支持pil读图

File: ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml (hunk 3 above)
Comment (flytocc) [reply ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 确实有点问题，已修改。
> update_freq: 4  # for 8 cards

total_batch_size = update_freq * num_gpus * batch_size_per_gpu

File: ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml (hunk 1 above)
Comment (flytocc) [reply ppcls/configs/ImageNet/ConvNeXt/convnext_tiny.yaml]: 改了

File: ppcls/configs/ImageNet/ConvNeXt/ConvNeXt_tiny.yaml [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  update_freq: 4  # for 8 cards
+
+# model ema
+EMA:
+  decay: 0.9999
+
+
+# model architecture
+Arch:
+  name: ConvNext_tiny
```
Comment (cuicheng01) [inline ppcls/configs/ImageNet/ConvNeXt/ConvNeXt_tiny.yaml]: 模型名字需要保持一致，建议统一成ConvNeXt_tiny

File: ppcls/data/preprocess/ops/operators.py (hunk 4 above)
Comment (cuicheng01) [reply ppcls/data/preprocess/ops/operators.py]: 现在也是支持的吧

File: ppcls/configs/ImageNet/ConvNeXt/ConvNeXt_tiny.yaml (hunk 5 above)
Comment (flytocc) [reply ppcls/configs/ImageNet/ConvNeXt/ConvNeXt_tiny.yaml]: done

**Review by cuicheng01 — APPROVED** [review]
